pinot-common/src/main/java/com/linkedin/pinot/common/utils/SegmentNameBuilder.java

```diff
@@ -51,34 +54,85 @@ public static String build(String groupId, String partitionRange, String sequenc
       return StringUtils.join(Lists.newArrayList(groupId, partitionRange, sequenceNumber), "__");
     }
 
+    public static String buildRealtimeV2Name(String tableName, String partitionId, String sequenceNumber,
+        long millisSinceEpoch) {
+      // ISO8601 date: 2016-01-20T123456.789Z
+      DateTime dateTime = new DateTime(millisSinceEpoch, DateTimeZone.UTC);
+      String date = dateTime.toString("yyyy-MM-DD'T'HHmmss.SSS'Z'");
+
+      // New realtime consumer v2 name: {tableName}__{partitionId}__{sequenceNumber}__{date}
+      return StringUtils.join(Lists.newArrayList(tableName, partitionId, sequenceNumber, date), "__");
+    }
+
     public static String extractTableName(String segmentId) {
-      if (!isOldStyleName(segmentId)) {
+      if (isOldV1StyleName(segmentId) || isRealtimeV2Name(segmentId)) {
+        return segmentId.split("__")[0];
+      } else if (isShortV1StyleName(segmentId)) {
         // Table name is the first part of the Kafka consumer group id
         String groupId = extractGroupIdName(segmentId);
         return groupId.substring(0, groupId.indexOf(REALTIME_SUFFIX) + REALTIME_SUFFIX_LENGTH);
       } else {
-        return segmentId.split("__")[0];
+        throw new RuntimeException("Unable to parse segment name " + segmentId);
       }
     }
 
     public static String extractGroupIdName(String segmentId) {
-      if (!isOldStyleName(segmentId)) {
+      if (isOldV1StyleName(segmentId)) {
+        return segmentId.split("__")[2];
+      } else if (isShortV1StyleName(segmentId)) {
         return segmentId.split("__")[0];
+      } else if (isRealtimeV2Name(segmentId)){
+        throw new RuntimeException("Realtime v2 segments don't have a consumer group id");
       } else {
-        return segmentId.split("__")[2];
+        throw new RuntimeException("Unable to parse segment name " + segmentId);
       }
     }
 
     public static String extractPartitionRange(String segmentId) {
-      if (!isOldStyleName(segmentId)) {
+      if (isOldV1StyleName(segmentId)) {
+        return segmentId.split("__")[3];
+      } else if (isShortV1StyleName(segmentId)) {
         return segmentId.split("__")[1];
+      } else if (isRealtimeV2Name(segmentId)){
+        return segmentId.split("__")[2];
       } else {
-        return segmentId.split("__")[3];
+        throw new RuntimeException("Unable to parse segment name " + segmentId);
       }
     }
 
-    private static boolean isOldStyleName(String segmentId) {
+    public static String extractSequenceNumber(String segmentId) {
+      if (isOldV1StyleName(segmentId)) {
```

Comment:
Actually, I am assuming in my code that there is an object from which can encode/decode segment names. If you can create a class that will be really useful for merging (and better as well for the long run).

e.g.

SegmentNameHolder(string name)
HighLevelSegmentNameHolder(components...)
LowLevelSegmentNameHolder(components...)
isHighLevel()
getterForComponent()

I think you get the idea.

If there are limited places where this is called, then please go ahead and change them, otherwise please create a ticket. 

Does this seem OK?